modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Plugin/GCIdentityProvider/Personify.php

```diff
@@ -0,0 +1,206 @@
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\openy_gc_auth\GCIdentityProviderPluginBase;
+use Drupal\personify\PersonifyClient;
+use Drupal\personify\PersonifySSO;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Personify SSO identity provider plugin.
+ *
+ * @GCIdentityProvider(
+ *   id="personify",
+ *   label = @Translation("Personify provider"),
+ *   config="openy_gc_auth.provider.personify"
+ * )
+ */
+class Personify extends GCIdentityProviderPluginBase {
+
+  /**
+   * Personify SSO service.
+   *
+   * @var \Drupal\personify\PersonifySSO
+   */
+  protected $personifySSO;
+
+  /**
+   * Personify Client service.
+   *
+   * @var \Drupal\personify\PersonifyClient
+   */
+  protected $personifyClient;
+
+  /**
+   * The Messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    ConfigFactoryInterface $config,
+    EntityTypeManagerInterface $entity_type_manager,
+    MessengerInterface $messenger,
+    PersonifySSO $personifySSO,
+    PersonifyClient $personifyClient
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $config, $entity_type_manager);
+    $this->messenger = $messenger;
+    $this->personifySSO = $personifySSO;
+    $this->personifyClient = $personifyClient;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('entity_type.manager'),
+      $container->get('messenger'),
+      $container->get('personify.sso_client'),
+      $container->get('personify.client')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration():array {
+    return [
+      'application_url' => '/gated-content',
+      'api_login_check' => '/openy_gc_auth_personify/check/login',
+      'api_auth' => '/openy_gc_auth_personify/auth',
+      'api_logout' => '/openy_gc_auth_personify/logout',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $config = $this->getConfiguration();
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    $personifyConfigForm = Link::createFromRoute($this->t('Personify configuration form'),
+      'personify.settings', [], ['target' => '_blank'])->toString();
+    $readMeUrl = Link::fromTextAndUrl($this->t('README file'),
+      Url::fromUserInput('/' . drupal_get_path('module', 'openy_gc_auth_personify') . '/README.md',
```

Comment:
`*.md` file extension should be enabled on the server level for downloading.

Reply:
I can change it to *.txt then or completely remove the link.
What will be more suitable and common for the OpenY?

Reply:
we could be using hook_help() in order to not introduce environmental issues
*.txt could work too, but not all webhost setups allowing these as well